crates/core/tedge/src/cli/certificate/cli.rs

```diff
@@ -97,7 +97,12 @@ impl BuildCommand for TEdgeCertCli {
                 let cloud: Option<Cloud> = cloud.map(<_>::try_into).transpose()?;
 
                 // Use the current device id if no id is provided
-                let id = id.unwrap_or(get_device_id_from_cloud(&config, &cloud)?);
+                let id = if let Some(id) = id {
```

Comment:
At first, I wrote this line as below.

```rust
let id = id.unwrap_or(get_device_id_from_cloud(&config, &cloud)?);
```

However, it causes a problem when certificate doesn't exist but id has `Some()` since`unwrap_or()` evaluates the default always regardless of id is `Some` or `None`.

Reply:
Indeed, and `unwrap_or_else`, makes it so you can't use `?` from within the closure, which is a bit of a pain.